FlowCrypt/Functionality/Mail Provider/Message Provider/MessageService.swift

```diff
@@ -288,6 +276,44 @@ final class MessageService {
     }
 }
 
+// MARK: - Message verification
+extension MessageService {
+    private func fetchVerificationPubKeys(for sender: String?) -> [String] {
+        if let sender = sender {
+            return contactsService.retrievePubKeys(for: sender)
+        } else {
+            return []
+        }
+    }
+
+    private func evaluateSignatureVerificationResult(
+        signature: MsgBlock.VerifyRes?,
+        sender: String?
+    ) async -> ProcessedMessage.MessageSignature {
+        guard let signature = signature else { return .unsigned }
+
+        if let error = signature.error {
+            return .error(error)
+        }
+
+        guard let signer = signature.signer else { return .unsigned }
+
+        var pubKey: PubKey?
+
+        if let contact = await contactsService.findBy(longId: signer) {
+            pubKey = contact.pubKey(with: signer)
+        } else if let email = sender, let contact = try? await contactsService.searchContact(with: email) {
+            pubKey = contact.pubKey(with: signer)
+        }
```

Comment:
Now I'm thinking, this shouldn't even be needed anymore - since we already provided the right pubkey to TypeScript. It would not come out as `match` unless it had the right pubkey to verify.

Therefore you can remove this `pubKey` code from this method, including `pubKey != nil` below. In fact, can also remove `sender` from this method signature.

Reply:
We can remove first part `contactsService.findBy(longId: signer)`, but second part `contactsService.searchContact(with: email)` is needed for the case when mail sender isn't stored in local user contacts and we need to fetch data using `pubLookup.lookup(with: email)`

For example, I reinstalled app and my local contacts list is empty, then verification for all messages will fail with error `no public key`

Or I'm missing something and there is another way to get sender's public key?

Reply:
I think I found a bug here - this lookup should be performed before message validation, if app doesn't have sender's pubkey. Fetching here is too late, as we've already tried to validate message.
I'll remove pubkey fetch here and move it to previous validation step.

Reply:
There are performance concerns - it may take too long. So I was thinking to fetch in parallel with downloading the message. That's why I did #1002 (pub lookup itself was slow) and also separated "fetch if pubkeys missing" part into #1001

But if you merge in #1002 and then also do #1001 in this PR, that would be ok with me. Otherwise can wait for later - up to you.

Reply:
I update `fetchVerificationPubKeys` function to fetch pub keys in case when there are no local pub keys - https://github.com/FlowCrypt/flowcrypt-ios/pull/998/commits/e304d431539c2d5bb7634672ca01ae110fae9e0d#diff-a1ec6a13dae98c2b61953533ba95947cc194b42dce60f4d2c905ef5ea9df1b64R280
So if keys are already stored locally then there won't be any delay for user.

Reply:
This works as a compromise. We'll have to do something better in the future.

Because public keys can change over time. So just because I have a pubkey for someone, doesn't mean that they don't have a new key (eg keys may get rotated once a year in some orgs) and that they didn't sign that message with the new key that I don't have.

But as it is now, it settles 99% of cases which will do for now.